src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1782,12 +1824,18 @@ bool AdaptDispatch::EnableAnyEventMouseMode(const bool enabled)
 // True if handled successfully. False otherwise.
 bool AdaptDispatch::EnableAlternateScroll(const bool enabled)
 {
-    // TODO GH#XXXX:
-    //   This is a temporary replacement to enable passhthrough
-    //   mode for Windows Terminal. Replace with proper _pConApi
-    //   call below when ConPty has been properly updated.
-    _pConApi->PrivateEnableAlternateScroll(enabled);
-    return false;
+    bool success = true;
+    success = _pConApi->PrivateEnableAlternateScroll(enabled);
+
+    // If we're a conpty, always return false
+    bool isPty = false;
+    _pConApi->IsConsolePty(isPty);
+    if (isPty)
+    {
+        return false;
+    }
```

Comment:
god, this API is needlessly verbose ;)

i hope when somebody fixes `IsConsolePty` to return a bool, they come fix all of these